File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        for (size_t i = 0; i <= nums.size(); i++)
        {

            if (nums.at(i) == target)
            {
                return i;
            }
            else if (nums.at(i) > target)
            {
                return i;
            }
            else if (i == nums.size() - 1)
            {
                return i + 1;
            }
        }
        return 0;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int index = 0;
        for (int i : nums) {
            if (i == target) {
                return index;
            }else if (i > target) {
                return index;
            }else if (index == nums.length - 1) {
                return index + 1;
            }
            index++;
        }
        return 0;
    }
}